language/bytecode_verifier/src/type_memory_safety.rs

```diff
@@ -59,7 +59,11 @@ impl<'a> TypeAndMemorySafetyAnalysis<'a> {
             } else {
                 locals.insert(
                     arg_idx as LocalIndex,
-                    AbstractValue::full_value(arg_type_view.kind()),
+                    AbstractValue::full_value(
+                        arg_type_view
+                            .kind(&function_definition_view.signature().as_inner().type_formals)
```

Comment:
nit, I would bind this to a local to make it easier to read :) 

It will probably reduce the line count